libvisual/CMakeLists.txt

```diff
@@ -179,14 +187,29 @@ ENDIF()
 
 # Git revision
 FIND_PACKAGE(Git)
-CONFIGURE_FILE(
-  "${PROJECT_SOURCE_DIR}/update-version.sh.in"
-  "${PROJECT_BINARY_DIR}/update-version.sh"
-  @ONLY
-)
-ADD_CUSTOM_TARGET(update-version ALL
-  COMMAND /bin/sh ${PROJECT_BINARY_DIR}/update-version.sh
-)
+
+EXECUTE_PROCESS(COMMAND ${GIT_EXECUTABLE} describe --always --dirty
+  OUTPUT_VARIABLE LV_REVISION OUTPUT_STRIP_TRAILING_WHITESPACE)
+IF(NOT LV_REVISION)
+  SET(LV_REVISION "unknown")
+ENDIF()
+
+IF(EXISTS ${BINARY_DIR}/version.h)
+  FILE(STRINGS ${BINARY_DIR}/version.h OLD_REVISION REGEX "#define LV_REVISION")
+  STRING(REGEX MATCH \".*\" OLD_REVISION ${OLD_REVISION})
+  STRING(REGEX MATCHALL "[^\"][0-9A-z]+" OLD_REVISION ${OLD_REVISION})
+ELSE()
+  SET(OLD_REVISION "")
+ENDIF()
+
+IF(NOT LV_REVISION EQUAL OLD_REVISION)
```

Comment:
This seems to be an optimisation to not re-generate a file with identical content every time someone runs CMake — true? If that's the case I wonder if that's worth the increase in complexity. If we keep it, let's add a comment why we need it.

Reply:
I'm not sure since that's what the original version.sh was doing so I was just replicating it. I'm not sure if it's necessary because I haven't seen someone use the same build folder contents between builds yet

Reply:
Okay